stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt

```diff
@@ -239,15 +240,19 @@ internal class ChatDomainImpl internal constructor(
             val initialSyncState = SyncStateEntity(currentUser.id)
             syncState = repos.syncState.select(currentUser.id) ?: initialSyncState
             // set active channels and recover
-            syncState?.let {
-                for (channelId in it.activeChannelIds) {
-                    channel(channelId)
-                }
-                // queries
-                val queries = repos.queryChannels.selectById(it.activeQueryIds)
-                for (queryChannelsSpec in queries) {
-                    queryChannels(queryChannelsSpec.filter, queryChannelsSpec.sort)
+            syncState?.let { state ->
+                // restore channels
+                state.activeChannelIds.forEach { channel(it) }
+
+                // restore queries
+                repos.queryChannels.selectById(state.activeQueryIds).forEach { spec ->
+                    queryChannels(spec.filter, spec.sort)
                 }
+
+                // retrieve the last time the user marked all as read and handle it as an event
+                state.markedAllReadAt
+                    ?.let { MarkAllReadEvent(user = currentUser, createdAt = it) }
+                    ?.let { eventHandler.handleEvent(it) }
```

Comment:
I'm not sure this works...

Reply:
so here's my concern:

- you trigger this event
- the client receives a new list of channels (which overwrites the reads)
- you end up showing channels that you marked as read, as unread